pkgs/by-name/am/amneziawg-tools/package.nix

```diff
@@ -0,0 +1,88 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  iptables,
+  iproute2,
+  makeWrapper,
+  openresolv,
+  procps,
+  bash,
+  amneziawg-go,
+  nix-update-script,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "amneziawg-tools";
+  version = "1.0.20240213";
+
+  src = fetchFromGitHub {
+    owner = "amnezia-vpn";
+    repo = "amneziawg-tools";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-QQtAEAsZ5O7gm++wum3rRV7BMaRZ6kP+0EjGE2iqcYw=";
+  };
+
+  sourceRoot = "${finalAttrs.src.name}/src";
+
+  outputs = [
+    "out"
+    "man"
+  ];
+
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ bash ];
+
+  makeFlags = [
+    "DESTDIR=${placeholder "out"}"
+    "PREFIX=/"
+    "WITH_BASHCOMPLETION=yes"
+    "WITH_SYSTEMDUNITS=yes"
+    "WITH_WGQUICK=yes"
+  ];
+
+  postFixup =
+    ''
+      substituteInPlace $out/lib/systemd/system/awg-quick@.service \
+        --replace-fail /usr/bin $out/bin
+    ''
+    + lib.optionalString stdenv.isLinux ''
+      for f in $out/bin/*; do
+        # Which firewall and resolvconf implementations to use should be determined by the
+        # environment, we provide the "default" ones as fallback.
+        wrapProgram $f \
+          --prefix PATH : ${
+            lib.makeBinPath [
+              procps
+              iproute2
+            ]
```

Comment:
When starting the `systemd` unit, `awg-quick` complained about `readlink` missing. I added `coreutils` here to resolve it.

Reply:
I checked awg-quick@name service, it doen't work, like the wg-quick@name. In [wg-quick module](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/networking/wg-quick.nix) some tools, like `config.networking.firewall.package` are added to service path. I think best solution is to update wireguard modules and add something like `type = wireguard | amneziawg` option.